src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -136,6 +140,23 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
                 let llfn = self.get_intrinsic(&("llvm.debugtrap"));
                 self.call(llfn, &[], None)
             }
+            "count_code_region" => {
+                if let ty::InstanceDef::Item(fn_def_id) = caller_instance.def {
+                    let caller_fn_path = tcx.def_path_str(fn_def_id);
+                    debug!(
+                        "count_code_region to llvm.instrprof.increment(fn_name={})",
+                        caller_fn_path
+                    );
+
+                    let (fn_name, _len_val) = self.const_str(Symbol::intern(&caller_fn_path));
+                    let index = args[0].immediate();
+                    let hash = self.const_u64(1234);
```

Comment:
The LLVM documentation is fairly vague about the `hash` value - is using `1234` intended to be permanent?

Reply:
No, definitely not. Thanks for asking. This is just a stand-in for now. The `fn_name` will also be changed to a mangled function name.

Reply:
Adding a FIXME would help avoid confusion.

Reply:
Agreed. Will do.

Reply:
Done